debian/authd.gdm-authd.pam

```diff
@@ -0,0 +1,27 @@
+#%PAM-1.0
+auth    [success=ok user_unknown=ignore default=bad] pam_succeed_if.so user != root quiet_success
+auth    [success=2 ignore=ignore default=die] pam_authd.so
```

Comment:
I'm not a big fan of using `default=die` in our module config as we could lock out the system entirely if `authd` can't be executed. Can we think about something better? 

cc. @didrocks

Reply:
Anyway, we are including common-auth just below, to authd will be already provided in common-auth, no? Also, the fact that we have `success=2`, hardcoded, where actually the amount of steps we want to jump over depends on the one in common-auth (if the content is only included in that one) seems fragile to me?

Reply:
> I'm not a big fan of using `default=die` in our module config as we could lock out the system entirely if `authd` can't be executed. Can we think about something better?

Well, if we fail, really we should be locked down when using gdm with authd mode. In case it fallback to local broker then we're ignoring, so that's handled.

In any case, in GDM you can technically always switch pam profile to use, so for example smart-cards or fingerprint profiles will always be running despite this failing.

> Anyway, we are including common-auth just below, to authd will be already provided in common-auth, no? Also, the fact that we have success=2, hardcoded, where actually the amount of steps we want to jump over depends on the one in common-auth (if the content is only included in that one) seems fragile to me?

Oh, yeah, in fact it seems I eventually pushed the old version of this file -_-, we've in fact to use `substack` to handle this in this way.

This file is basically based on `gdm-smartcard-sssd-exclusive` pam config that I wrote some time ago and it's quite well tested by autopkgtests too.

An alternative to avoid using jumps (but repetitions) is instead to do something like this:

```
#%PAM-1.0
auth    [success=ok user_unknown=ignore default=bad] pam_succeed_if.so user != root quiet_success
auth    [success=ok ignore=3 default=die] pam_authd.so
auth    requisite       pam_nologin.so
auth    optional        pam_gnome_keyring.so
auth    sufficient      pam_permit.so

# If authd is ignoring the request we fallback to default auth
@include common-auth
auth    requisite       pam_nologin.so
auth    optional        pam_gnome_keyring.so
```

Reply:
I think this is more reasonable if we don’t include common-auth and do the same than other gdm specific processes.

Reply:
> I think this is more reasonable if we don’t include common-auth and do the same than other gdm specific processes, so +1 on the correct file (happy to have spotted it!)

No check the edit, it was fine already :-D. For some reason I thought the pushed file was using `@include` but it didn't, it was using `substack` and so was fine.

And IMHO it's better to use it compared to including.

Reply:
Mostly because we'd be more conformant to what we already do for other gdm stacks, and being consistent with the others, makes it easier to debug or address potential issues.

Reply:
Ah, so the fix was to change the `success=2` to 1 :)

Reply:
agreed on following the rest of the gdm stacks.

Reply:
> Ah, so the fix was to change the `success=2` to 1 :)

No, it was already working as this, but I noticed that _being consistent_ I had copied too much from gdm one :-D, were we do an extra step and so we've to skip 2 times.

By having `2` here we'd missed using `pam_nologin.so` which isn't super-needed but still something we should do

Reply:
got it!